process/coordinator/blockDataRequest_test.go

```diff
@@ -154,6 +234,683 @@ func TestTransactionCoordinator_requestMissingMiniBlocksAndTransactionsShouldWor
+		require.Nil(t, err)
+		require.True(t, timeFunctionCalled)
+	})
+}
+
+func TestBlockDataRequest_receivedMiniBlock(t *testing.T) {
+	t.Parallel()
+
+	t.Run("nil key should return early", func(t *testing.T) {
+		t.Parallel()
+		args := createMockArgs()
+		blockDataRequester, _ := NewBlockDataRequester(args)
+
+		// Should not panic
+		blockDataRequester.receivedMiniBlock(nil, &block.MiniBlock{})
+	})
+
+	t.Run("unrequested mini block should return early", func(t *testing.T) {
+		t.Parallel()
+		args := createMockArgs()
+		blockDataRequester, _ := NewBlockDataRequester(args)
+
+		// Mini block was not requested
+		key := []byte("hash1")
+		miniBlock := &block.MiniBlock{
+			Type:     block.TxBlock,
+			TxHashes: [][]byte{[]byte("tx1"), []byte("tx2")},
+		}
+
+		// Should not panic and should return early
+		blockDataRequester.receivedMiniBlock(key, miniBlock)
+	})
+
+	t.Run("wrong type assertion should log warning and return", func(t *testing.T) {
+		t.Parallel()
+		args := createMockArgs()
+		blockDataRequester, _ := NewBlockDataRequester(args)
+
+		// Add the mini block to requested items
+		key := []byte("hash1")
+		_ = blockDataRequester.requestedItemsHandler.Add(string(key))
+
+		// Pass wrong type (not a mini block)
+		wrongValue := "not a mini block"
+
+		// Should not panic and should log warning
+		blockDataRequester.receivedMiniBlock(key, wrongValue)
+	})
+
+	t.Run("preprocessor not found should log warning and return", func(t *testing.T) {
+		t.Parallel()
+		args := createMockArgs()
+		blockDataRequester, _ := NewBlockDataRequester(args)
+
+		// Add the mini block to requested items
+		key := []byte("hash1")
+		_ = blockDataRequester.requestedItemsHandler.Add(string(key))
+
+		// Create mini block with type that has no preprocessor
+		miniBlock := &block.MiniBlock{
+			Type:     block.InvalidBlock, // This type won't have a preprocessor
+			TxHashes: [][]byte{[]byte("tx1")},
+		}
+
+		// Should not panic and should log warning
+		blockDataRequester.receivedMiniBlock(key, miniBlock)
+	})
+
+	t.Run("should successfully request transactions for mini block", func(t *testing.T) {
+		t.Parallel()
+		args := createMockArgs()
+
+		// Create preprocessors container with mock
+		preprocContainer := containers.NewPreProcessorsContainer()
+		txPreproc := &preprocMocks.PreProcessorMock{
+			RequestTransactionsForMiniBlockCalled: func(miniBlock *block.MiniBlock) int {
+				return len(miniBlock.TxHashes)
+			},
+		}
+		_ = preprocContainer.Add(block.TxBlock, txPreproc)
+		args.PreProcessors = preprocContainer
+
+		blockDataRequester, _ := NewBlockDataRequester(args)
+
+		// Add the mini block to requested items
+		key := []byte("hash1")
+		_ = blockDataRequester.requestedItemsHandler.Add(string(key))
+
+		// Create mini block
+		miniBlock := &block.MiniBlock{
+			Type:     block.TxBlock,
+			TxHashes: [][]byte{[]byte("tx1"), []byte("tx2"), []byte("tx3")},
+		}
+
+		// Should process successfully
+		blockDataRequester.receivedMiniBlock(key, miniBlock)
```

Comment:
no check for panic.

should also check that the request transactions for the miniblock was called successfully.